lib/api/v3/work_packages/schema/work_package_schema.rb

```diff
@@ -63,19 +59,19 @@ def assignable_statuses_for(user)
           end
 
           def assignable_versions
-            @work_package.assignable_versions if defines_assignable_values?
+            @work_package.assignable_versions unless @work_package.nil?
           end
 
           def assignable_priorities
-            @work_package.assignable_priorities if defines_assignable_values?
+            IssuePriority.where(active: true)
           end
 
           def assignable_categories
-            @work_package.assignable_categories if defines_assignable_values?
+            project.categories
           end
 
           def available_custom_fields
-            @project.all_work_package_custom_fields & @type.custom_fields.all
+            project.all_work_package_custom_fields & type.custom_fields.all
```

Comment:
@NobodysNightmare will this conflict with what you are doing in the other PR right now? Any preference of merging order?